packages/xrpl/src/Wallet/index.ts

```diff
@@ -441,7 +441,11 @@ export class Wallet {
         ? decode(signedTransaction)
         : signedTransaction
     const messageHex: string = encodeForSigning(tx)
-    const signature = tx.TxnSignature
+    // Need a SignedTransaction class where TxnSignature is not optional.
+    // verify signature used to be untyped. `tx` is union of Transaction and
+    // JsonObject. Maybe decode should take a type parameter?
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- See above
+    const signature = tx.TxnSignature as string
```

Comment:
TODO @ckniffen

Reply:
We should throw an exception if it is not set.

Reply:
That's much better and simple

Reply:
TODO

Reply:
Has this TODO been resolved?

Reply:
bump